classes/local/factor/object_factor_base.php

```diff
@@ -59,21 +59,39 @@ public function __construct($name) {
 
         // Setup secret manager.
         $this->secretmanager = new \tool_mfa\local\secret_manager($this->name);
+    }
+
+    /**
+     * This loads the locked state from the DB
+     *
+     * Base class implementation.
+     *
+     */
+    public function check_locked_state() {
```

Comment:
As this doesnt really return anything, its name might be better as process_locking_state or similar?